tests/e2e/scenarios/test_slack_e2e.py

```
@@ -149,6 +150,44 @@ async def activate_slack(
+    base_url: str,
+    *,
+    token: str,
+    owner_id: str,
+    channels_dir: str,
+) -> None:
+    """Install and configure Slack without fake-provider reset/pairing helpers."""
+    await install_slack(base_url)
+    _patch_slack_capabilities_for_testing(channels_dir)
+
+    cap_path = os.path.join(channels_dir, "slack.capabilities.json")
+    with open(cap_path, "r") as f:
```

Ruff (0.15.18)

[warning] Async functions should not open files with blocking methods like `open`

(ASYNC230)